style.css

```diff
@@ -1,8 +1,169 @@
-#app {
-  font-family: Avenir, Helvetica, Arial, sans-serif;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
+@import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:wght@500&family=Rubik:wght@600&display=swap');
+
+
+:root {
+  --primary-color: #026a90;
+  --secondary-color: lightgray;
+}
+
+* {
+  box-sizing: border-box;
+
+}
+
+body {
+  background-color: #FFFFFF;
+  font-family: sans-serif;
+  margin: 0;
+}
+
+header{
+  margin: 0 auto;
+  width: 100%;
   text-align: center;
-  color: #2c3e50;
-  margin-top: 60px;
 }
+h1{
+  margin: 0 0 50px;
+  font-size: 48px;
+  font-weight: 400;
+
+
+font-family: 'Roboto Mono', monospace;
+  color: #000;
+}
+
+.search {
+  background-image: url('search.png');
+  background-size: 20px;
+  background-position: calc(100% - 10px) center;
+  background-repeat: no-repeat;
+  background-color: white;
+  border: 1px solid black;
+  border-radius: 0px;
+  font-size: 1rem;
+  padding: 0.5rem;
+  color: black;
+}
+
+
+.search::placeholder {
+  color: black;
+}
+.search:focus::placeholder {
+  color: transparent;
+}
+
+.search:focus {
+  outline: none;
+}
+
+
+#main{
+  display: flex;
+  justify-content: center;
+  max-width: 1200px;
+  width:100%;
+  margin: 80px auto 0;
+  flex-wrap: wrap;
+  gap:48px 75px;
+}
+
+.card {
+  display: block;
+  cursor: pointer;
+  background-position: center;
+  background-size: cover;
+  background-repeat: no-repeat;
+position: relative;
+
+height: 374px;
+width: 350px;
+box-shadow: 0px 10px 20px 5px rgba(0, 0, 0, 0.25);
+border-radius: 20px;
+}
+
+
+
+.title{
+  position: absolute;
+  left: 50%;
+  
+  top: 50%;
+  transform: translate(-50%,-50%);
+  
+  font-family: 'Rubik';
+  font-weight: 600;
+  font-size: 24px;
+  line-height: 28px;
+  letter-spacing: 0.004em;
+  color: #FFFFFF;
+}
+
+
+
+.button{
+  position: absolute !important;
+  left:50%;
+  transform: translateX(-50%);
+  
+  bottom: 56px;
+  font-family: 'Rubik';
+  font-style: normal;
+  font-weight: 500;
+  font-size: 16px;
+  line-height: 19px;
+  
+  background-color: #352C9A;
+  color: white;
+  min-width :145px; ;
+  height: 43px;
+  border-style: none;
+}
+
+/* .card {
+  content: '';
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  transition: 250ms ease-out;
+  border-radius: 20px;
+} */
+.card button {
+  color: #fff;
+  opacity: 0;
+  position: relative;
+  z-index: 2;
+  transition: 250ms ease-out;
+}
+.card h3 {
+  color: #fff;
+  opacity: 0;
+  
+  z-index: 2;
+  transition: 250ms ease-out;
+}
+.card::after { 
+  border-radius: 20px;
+  opacity: 0; 
+  top:0;
+  left:0;
+  right:0;
+
+  visibility: 0;
+  position: absolute; 
+  width: 100%; height: 100%; background-color: rgba(98,98,98,0.6); content:'';}
+
+.card:hover .button {
+  display: block;
```

Comment:
Transitioning from opacity 0 to opacity 1 will be enough in this case, there is no need to use display: block. Using display black will affect the transition of the opacity.